torch/custom_libraries.py

```diff
@@ -5,27 +5,36 @@
 # User created fragment libraries to extend existing libraries
 fragments_for_existing_libraries = {}
 
+# User defined libraries
+libraries = {}
+
 # Set containing the combination of (namespace, operator, DispatchKey) for which a new kernel has been registered
 # The keys in the set are of the form `namespace + "/" + op_name + "/" + dispatch_key`.
 # This set is maintained to ensure that two libraries don't try to override the exact same functionality to avoid
 # libraries calling into kernels not intended to be called.
 impls = set()
 
 class _Library:
-    # kind can be DEF, FRAGMENT (in C++ IMPL)
+    # kind can only be FRAGMENT for now.
+    # TODO: add support for DEF
     def __init__(self, kind, ns, dispatch_key=""):
        # also prevent users from creating libraries with existing C++ libraries, e.g., aten
-       if ns in existing_libraries and kind == "DEF":
-          raise ValueError("A library with name '" + ns + "' already exists. " \
-                            "It's not allowed to have more than one library with the same name")
```

Comment:
Since you're modifying code that you added in the previous PR in the stack, I suggest just smooshing the two diffs together and doing them all in one go. Stack management is going to be a pain otherwise.